weaviate/logger.py

```diff
@@ -1,5 +1,250 @@
+    
+    print(f"DEBUG: Logger initialized with level {logger.level}, handlers: {logger.handlers}")
+    return logger
+
+logger = _setup_logger()
+
+def _mask_sensitive_headers(headers: Dict[str, str], is_response: bool = False) -> Dict[str, str]:
+    """Mask sensitive information in headers like authorization tokens and cookies.
+    
+    Args:
+        headers: Dictionary of headers to mask
+        is_response: Whether these are response headers (affects which headers to mask)
+    
+    Returns:
+        Dictionary with sensitive headers masked
+    """
+    masked = {}
+    for key, value in headers.items():
+        key_lower = key.lower()
+        if key_lower == "authorization" or "authorization" in key_lower:
+            value = "[...]"
+        elif key_lower == "cookie" and not is_response:
+            value = value.split("=")[0] + "=..."
+        elif key_lower == "set-cookie" and is_response:
+            value = value.split("=")[0] + "=..."
+        elif any(api_key in key_lower for api_key in ["api-key", "api_key", "token", "secret"]):
+            value = "[...]"
+        elif "access-control-allow-headers" in key_lower and "authorization" in value.lower():
+            # Special handling for CORS headers that list Authorization
+            value = "[...]"
+        masked[key] = value
+    return masked
+
+def log_http_event(response: httpx.Response) -> None:
+    """Log HTTP request and response details using the default logger.
+    
+    Args:
+        response: The httpx Response object containing both request and response info
+    """
+    print(f"\nDEBUG: log_http_event called")
+    print(f"DEBUG: Logger state: level={logger.level}, handlers={logger.handlers}, propagate={logger.propagate}")
+    print(f"DEBUG: Effective level: {logger.getEffectiveLevel()}")
+
+    if logger.getEffectiveLevel() > logging.DEBUG:
+        print("DEBUG: Skipping logging due to level")
+        return
+
+    try:
+        request = response.request
+        print(f"DEBUG: Processing request {request.method} {request.url}")
+        
+        # Log request details with clear separation
+        separator = "==================== HTTP Request ===================="
+        print(f"DEBUG: About to log separator: {separator}")
+        logger.debug(separator)
+        
+        request_line = f"Request: {request.method} {request.url}"
+        print(f"DEBUG: About to log request line: {request_line}")
+        logger.debug(request_line)
+        
+        print("DEBUG: About to log headers")
+        logger.debug("Headers:")
+        masked_req_headers = _mask_sensitive_headers(dict(request.headers))
+        for key, value in masked_req_headers.items():
+            header_line = f"  {key}: {value}"
+            print(f"DEBUG: About to log header: {header_line}")
+            logger.debug(header_line)
+        
+        # Log request body if present
+        if request.content:
+            print("DEBUG: About to log request body")
```

Comment:
This is a `print` statement and not `logger.debug` and so will show up regardless of the `WEAVIATE_LOG_LEVEL` specification making the client noisy